### DISCUSSION TITLE
Boolean union operation of groups containing hundreds of thousands of elements.

### DISCUSSION THREAD

**Discussion by GiulioBj** [CadQuery/cadquery#1686] [Q&A]
I'm using CadQuery to perform a boolean operation on hundreds of thousands of small parallelepipeds, but the process is extremely slow. How can I speed it up as much as possible?

**Comment by adam-urbanczyk** [comment]
I don't think you can do a lot, probably you need to use a different (i.e. CSG) kernel. Are you using `cq.occ_impl.shapes.fuse`?

**Reply by GiulioBj** [reply]
I don't have that advanced level of programming to understand this thing. I assume yes if it is the kernel used by cadquery.

**Reply by adam-urbanczyk** [reply]
OK, could you share the code that you are using?
